elide-graphql/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -390,6 +410,28 @@ private PersistentResource upsertObject(Environment context, Entity entity) {
         return updateAttributes(upsertedResource, entity, attributes);
     }
 
+    private PersistentResource updateObject(Environment context, Entity entity) {
+        Set<Entity.Attribute> attributes = entity.getAttributes();
+        Optional<String> id = entity.getId();
+        RequestScope requestScope = entity.getRequestScope();
+        PersistentResource updatedResource;
+
+        if (!id.isPresent()) {
+            throw new BadRequestException("UPDATE data objects must include ids");
+        } else {
+            Set<PersistentResource> loadedResource = fetchObject(context, requestScope, entity.getEntityClass(),
+                Optional.of(Arrays.asList(id.get())),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                false).getPersistentResources();
+            updatedResource = loadedResource.iterator().next();
```

Comment:
what happens if this list is empty?

Reply:
The underlying call to fetchObject will throw an exception (InvalidObjectIdentifierException) so that will never happen.